pkg/cri/resource-manager/metrics/metrics.go

```diff
@@ -193,3 +196,16 @@ func dump(prefix string, f *model.MetricFamily) {
 func metricsError(format string, args ...interface{}) error {
 	return fmt.Errorf("metrics: "+format, args...)
 }
+
+// set the updated policy metrics.
+func (m *Metrics) Set(pm *metrics.PolicyMetrics) {
+	m.Lock()
+	defer m.Unlock()
+	m.policyMetrics = pm
+	data, err := json.Marshal(m.policyMetrics)
```

Comment:
nitpick: I think there is no reason to do the heavy lifting of `json.Marshal()` with that lock held. You could first marshal, then take the lock only for updating the cached data.